### DISCUSSION TITLE
type in myconfig.env

### DISCUSSION THREAD

**Discussion by monojk** [tillsteinbach/VWsFriend#30] [Q&A] [answered]
I seem to have problems retrieving the data for my e-up! as grafana shows "No Data" for "Battery" and the other widgets. 

weconnect-cli works for me and in the "Adapter configuration: vw-connect.0" of VWsFriend I see the field "Type" that I set to "VWv2" to avoid an error message from the log. Initially it was set to "ID". I also tried "VW" but that doesn't work either. I don't get a login error message.

What should I set in my myconfig.env ?

**Comment by tillsteinbach** [comment]
It looks like you are still on an old version containing iobroker. I completely removed that with 0.3.0 as a standalone solution allows me much more features. Would be great if you could check with a new installation of the latest release

**Comment by monojk** [comment]
I did a pull and the started again. But still no data in grafana....

**Reply by tillsteinbach** [reply]
What does the console say? Any errors?

**Comment by monojk** [comment]
No errors:
![vwfriends-log](https://user-images.githubusercontent.com/25121142/131876889-ba948462-d76f-44da-8e14-3139b5658175.jpg)

**Reply by tillsteinbach** [reply]
Somehow you still ended up in an installation with ioBroker. But the newer versions are not working with ioBroker anymore. See here: https://github.com/tillsteinbach/VWsFriend/blob/main/docker-compose.yml

**Comment by monojk** [comment]
Looks even less working after purging the docker installs and starting again (the port 4000 isn't responding and grafana shows "InfluxDB Error: Bad Gateway", the paths shown below for grafana claiming not readable/writable don't even exist):
>pi@rp4b:~ $ docker-compose --env-file ./myconfig.env up
Starting pi_vwsfriend_1  ... done
Starting pi_postgresdb_1 ... done
Starting pi_grafana_1    ... done
Attaching to pi_vwsfriend_1, pi_postgresdb_1, pi_grafana_1
postgresdb_1  | 
postgresdb_1  | PostgreSQL Database directory appears to contain a database; Skipping initialization
postgresdb_1  | 
grafana_1     | GF_PATHS_CONFIG='/etc/grafana/grafana.ini' is not readable.
grafana_1     | GF_PATHS_DATA='/var/lib/grafana' is not writable.
grafana_1     | GF_PATHS_HOME='/usr/share/grafana' is not readable.
grafana_1     | You may have issues with file permissions, more information here: http://docs.grafana.org/installation/docker/#migrate-to-v51-or-later
postgresdb_1  | 2021-09-09 11:32:36.478 UTC [1] LOG:  starting PostgreSQL 13.4 (Debian 13.4-1.pgdg100+1) on arm-unknown-linux-gnueabihf, compiled by gcc (Debian 8.3.0-6) 8.3.0, 32-bit
vwsfriend_1   | Fatal Python error: init_interp_main: can't initialize time
vwsfriend_1   | Python runtime state: core initialized
vwsfriend_1   | PermissionError: [Errno 1] Operation not permitted
vwsfriend_1   | 
vwsfriend_1   | Current thread 0xb6f7d460 (most recent call first):
vwsfriend_1   | <no Python frame>
postgresdb_1  | 2021-09-09 11:32:36.478 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
postgresdb_1  | 2021-09-09 11:32:36.479 UTC [1] LOG:  listening on IPv6 address "::", port 5432
postgresdb_1  | 2021-09-09 11:32:36.485 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
postgresdb_1  | 2021-09-09 11:32:36.500 UTC [14] LOG:  database system was shut down at 2021-09-09 11:32:24 UTC
postgresdb_1  | 2021-09-09 11:32:36.524 UTC [1] LOG:  database system is ready to accept connections
pi_vwsfriend_1 exited with code 1

**Reply by tillsteinbach** [reply]
Did you also delete the volumes when you purged the docker images? Did you make any changes to the docker-compose.yml? /var/lib/grafana should be writeable for example.
You are using a raspberry pi right? Maybe I should try to get one to debug the images here.

**Reply by monojk** [reply]
I'm not really familiar with docker. I ran this command for my cleanup: `docker image prune -a`
I didn't change the docker-compose.yml
Yes, I use a Raspberry 4B
>ls /var/lib/grafana
ls: cannot access '/var/lib/grafana': No such file or directory

**Reply by tillsteinbach** [reply]
You can do a `docker-compose down -v` this will also delete the volumes. or if you want to use docker commands: `docker volume ls` and then `docker volume rm <name of docker volume>`
But it looks still like there are somehow permission problems on your Raspberry.

**Comment by monojk** [comment]
Same problem after cleanup and `docker-compose down -v `

**Comment by tillsteinbach** [comment]
I found something about the permission problem: https://community.home-assistant.io/t/migration-to-2021-7-fails-fatal-python-error-init-interp-main-cant-initialize-time/320648

suggestion is to upgrade libseccomp2

**Reply by monojk** [reply]
Thanks, with this the access errors disappear, but now I get:
>vwsfriend_1   | Traceback (most recent call last):
vwsfriend_1   |   File "/opt/venv/bin/vwsfriend", line 5, in <module>
vwsfriend_1   |     from vwsfriend.vwsfriend import main
vwsfriend_1   |   File "/opt/venv/lib/python3.9/site-packages/vwsfriend/vwsfriend.py", line 16, in <module>
vwsfriend_1   |     from weconnect import weconnect
vwsfriend_1   |   File "/opt/venv/lib/python3.9/site-packages/weconnect/weconnect.py", line 18, in <module>
vwsfriend_1   |     from weconnect.elements.vehicle import Vehicle
vwsfriend_1   |   File "/opt/venv/lib/python3.9/site-packages/weconnect/elements/vehicle.py", line 9, in <module>
vwsfriend_1   |     from weconnect.elements.generic_status import GenericStatus
vwsfriend_1   |   File "/opt/venv/lib/python3.9/site-packages/weconnect/elements/generic_status.py", line 11, in <module>
vwsfriend_1   |     from weconnect.util import robustTimeParse, toBool
vwsfriend_1   |   File "/opt/venv/lib/python3.9/site-packages/weconnect/util.py", line 9, in <module>
vwsfriend_1   |     from PIL import Image  # type: ignore
vwsfriend_1   |   File "/opt/venv/lib/python3.9/site-packages/PIL/Image.py", line 114, in <module>
vwsfriend_1   |     from . import _imaging as core
vwsfriend_1   | ImportError: libjpeg.so.8: cannot open shared object file: No such file or directory
pi_vwsfriend_1 exited with code 1

**Reply by tillsteinbach** [reply]
hmmm, I'm a bit helpless :( I think I need to get my hands on a PI to be able to properly debug, or we find another user here.

**Reply by tillsteinbach** [reply]
Trying some more remote debugging :)

Can you try to replace:
`image: tillsteinbach/vwsfriend:latest`
with
`image: tillsteinbach/vwsfriend:experimental`
and give it another try?

**Comment by monojk** [comment]
cleaned up everything and use experimental in the yml file I get:
>vwsfriend_1   | usage: vwsfriend [-h] [--version] [-u USERNAME] [-p PASSWORD] [--netrc NETRC]
vwsfriend_1   |                  [-i INTERVAL] [--tokenfile TOKENFILE] [--no-token-storage]
vwsfriend_1   |                  [-v] [--config-dir CONFIGDIR] [--demo DEMO] [--with-database]
vwsfriend_1   |                  [--database-url DBURL] [--with-abrp] [--with-homekit]
vwsfriend_1   | vwsfriend: error: argument -u/--username: expected one argument

**Reply by tillsteinbach** [reply]
Hmm, are you sure this is the only thing you changed? Did you maybe forget the `--env-file yourconfig`?

**Reply by monojk** [reply]
I used `docker-compose --env-file ./myconfig.env up` and only changed to `image: tillsteinbach/vwsfriend:latest`

**Reply by tillsteinbach** [reply]
Hmmm, it is the other way round, replacing latest with experimental, but this is probably what you did.
I can try the experimental branch tomorrow on my machine again, but it was successfully used yesterday. This is what makes me wonder.
The error message makes me believe that the Variables for User and password are not passed from the env file to the container.

**Reply by tillsteinbach** [reply]
I reworked the yml a bit. You can now leave it untouched and just add VWSFRIEND_VERSION=experimental to your config

**Comment by monojk** [comment]
In answer to "_I reworked the yml a bit. You can now leave it untouched and just add VWSFRIEND_VERSION=experimental to your config_"

Now I get a different but similar error:
>vwsfriend_1   | usage: vwsfriend [-h] [--version] [-u USERNAME] [-p PASSWORD] [--netrc NETRC]
vwsfriend_1   |                  [-i INTERVAL] [--tokenfile TOKENFILE] [--no-token-storage]
vwsfriend_1   |                  [-v] [--config-dir CONFIGDIR] [--demo DEMO] [--with-database]
vwsfriend_1   |                  [--database-url DBURL] [--with-abrp] [--with-homekit]
vwsfriend_1   | vwsfriend: error: argument -i/--interval: Must be a number not smaller than 300
pi_vwsfriend_1 exited with code 2

I tried with VWCONNECT_INTERVAL=15 and also 300 giving the same error message

**Reply by tillsteinbach** [reply]
I see the problem. You still have the config from the initial version of VWsFriend. Since changing from iobroker to my own implementation the variable names changed to:
```
WECONNECT_USER=test@test.de
WECONNECT_PASSWORD=secret
WECONNECT_PIN=1234 # Pin is only necessary for some functions, if your car does not support these, the pin can be ignored
WECONNECT_INTERVAL=300
```
So if you replace VWCONNECT to WECONNECT you should get one step further.

**Comment by monojk** [comment]
Yes, that got me further, but still no data and 2 types of error:
>vwsfriend_1   | Traceback (most recent call last):
vwsfriend_1   |   File "/opt/venv/bin/vwsfriend", line 8, in <module>
vwsfriend_1   |     sys.exit(main())
vwsfriend_1   |   File "/opt/venv/lib/python3.9/site-packages/vwsfriend/vwsfriend.py", line 139, in main
vwsfriend_1   |     weConnect = weconnect.WeConnect(username=username, password=password, tokenfile=tokenfile,
vwsfriend_1   |   File "/opt/venv/lib/python3.9/site-packages/weconnect/weconnect.py", line 144, in __init__
vwsfriend_1   |     self.login()
vwsfriend_1   |   File "/opt/venv/lib/python3.9/site-packages/weconnect/weconnect.py", line 366, in login
vwsfriend_1   |     data = tokenResponse.json()
vwsfriend_1   |   File "/opt/venv/lib/python3.9/site-packages/requests/models.py", line 910, in json
vwsfriend_1   |     return complexjson.loads(self.text, **kwargs)
vwsfriend_1   |   File "/usr/lib/python3.9/json/__init__.py", line 346, in loads
vwsfriend_1   |     return _default_decoder.decode(s)
vwsfriend_1   |   File "/usr/lib/python3.9/json/decoder.py", line 337, in decode
vwsfriend_1   |     obj, end = self.raw_decode(s, idx=_w(s, 0).end())
vwsfriend_1   |   File "/usr/lib/python3.9/json/decoder.py", line 355, in raw_decode
vwsfriend_1   |     raise JSONDecodeError("Expecting value", s, err.value) from None
vwsfriend_1   | json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)

> postgresdb_1  | 2021-09-16 07:32:11.587 UTC [116] ERROR:  relation "settings" does not exist at character 27
postgresdb_1  | 2021-09-16 07:32:11.587 UTC [116] STATEMENT:  SELECT vwsfriend_url FROM settings LIMIT 1
postgresdb_1  | 2021-09-16 07:32:11.587 UTC [117] ERROR:  relation "vehicles" does not exist at character 88
postgresdb_1  | 2021-09-16 07:32:11.587 UTC [117] STATEMENT:  SELECT "vin" || ':' || "nickname" || ' (' || "model" || ' - ' || "carType" || ')' FROM vehicles

**Reply by tillsteinbach** [reply]
Ok, first of all: Good news is that there are now all libraries for the raspberry pi in the image. So I can merge the changes of the experimental branch.
Bad news is: WeConnect does some strange things on your login. I will try to add more debug output and then we can try again. In the meantime you could also restart the container and see if that fixes the problem. Right now the weconnect server is a bit instable.

**Comment by monojk** [comment, marked as answer]
Good news: after restarting everything works now. Many thanks!

**Reply by tillsteinbach** [reply]
Good to hear that! And thanks for sharing your problem, this will make the software more stable!

**Reply by tillsteinbach** [reply]
With release 0.5.3 the necessary changes are in the main branch and thus "experimental" is not required anymore.
I want to suggest that you remove `VWSFRIEND_VERSION=experimental` from the config again to get future updates
